src/node/coverage.ts

```diff
@@ -0,0 +1,267 @@
+          detail:
+            "If you don't want to do this right now, you can disable Codecov line coverage in the Codecov extension's settings.",
+        },
+        "Set an API Key"
+      );
+
+      if (result === "Set an API Key") {
+        apiKey = await window.showInputBox({
+          title: "Enter your Codecov API Key",
+          prompt:
+            "You can generate an API key in your account settings within the Codecov app.",
+        });
+        if (apiKey) await context.secrets.store("api.key", apiKey);
+        updateDecorations();
+        return;
+      }
+    }
+
+    if (!apiUrl) {
+      // Just reset it to default for this workspace
+      await config.update("api.url", "https://api.codecov.io", false);
+      apiUrl = "https://api.codecov.io";
+    }
+
+    const path = encodeURIComponent(
+      workspace.asRelativePath(activeEditor.document.fileName)
+    );
+
+    const pathToWorkspace = workspace.getWorkspaceFolder(
+      Uri.file(activeEditor.document.fileName)
+    )?.uri.path;
+
+    const gitConfig = Uri.file(`${pathToWorkspace}/.git/config`);
+    const remote = await workspace.fs
+      .readFile(gitConfig)
+      .then((buf) => buf.toString())
+      .then((string) => string.split("\n"))
+      .then((lines) => lines.find((line) => line.match(/git@.*:.*\/.*.git$/)))
+      .then((line) => line?.replace(/.*:/, "").replace(".git", "").split("/"));
+    if (!remote) return;
+    const [owner, repo] = remote;
+
+    const gitHead = Uri.file(`${pathToWorkspace}/.git/HEAD`);
+    const branch = await workspace.fs
+      .readFile(gitHead)
+      .then((buf) => buf.toString())
+      .then((string) => string.replace("ref: refs/heads/", "").slice(0, -1));
+
+    if (!branch) return;
+
+    // Don't need this right now, but may be useful in the future if we want to cache coverage
+    //const gitRefFile = Uri.file(`${pathToWorkspace}/.git/refs/heads/${branch}`);
+    //const commitHash = await workspace.fs
+    //  .readFile(gitRefFile)
+    //  .then((buf) => buf.toString());
+    //if (!commitHash) return;
+
+    const coverageUrl = `${apiUrl}/api/v2/${provider}/${owner}/repos/${repo}/file_report/${path}`;
+
+    // First try getting coverage for this branch
+    let error = null;
+    let coverage: Coverage = await axios
+      .get(`${coverageUrl}?branch=${encodeURIComponent(branch)}`, {
+        headers: {
+          accept: "application/json",
+          authorization: `Bearer ${apiKey}`,
+        },
+      })
+      .then((response) => response.data)
+      .catch((error) => {
+        if (error?.response?.status >= 500) {
+          window.showErrorMessage(
+            "Codecov: Unable to connect to server or something went seriously wrong. Double check your API key in the Codecov extension settings"
+          );
+        } else if (error?.response.status === 401) {
+          window.showErrorMessage(
+            "Codecov: The provided API key is unauthorized to access this repository."
```

Comment:
Is there a link we can direct the user to with this message to properly fetch their key?